src/_gettsim/policy_environment.py

```diff
@@ -298,21 +293,13 @@ def load_functions_for_date(date):
 
     """
     year = date.year
-    functions = {}
-    if year < 2009:
-        functions["sum_eink"] = sum_eink_mit_kapital
-    else:
-        functions["sum_eink"] = sum_eink_ohne_kapital
 
-    if year <= 2014:
-        functions["alleinerz_freib_tu"] = eink_st_alleinerz_freib_tu_bis_2014
-    else:
-        functions["alleinerz_freib_tu"] = eink_st_alleinerz_freib_tu_ab_2015
-
-    if year <= 2004:
-        functions["eink_st_altersfreib"] = eink_st_altersfreib_bis_2004
-    else:
-        functions["eink_st_altersfreib"] = eink_st_altersfreib_ab_2005
+    # Using TIME_DEPENDENT_FUNCTIONS here leads to failing tests.
+    functions = {
+        f.__info__["dates_active_dag_key"]: f
+        for f in load_internal_functions().values()
+        if is_time_dependent(f) and is_active_at_date(f, date)
+    }
 
     if year <= 1996:
```

Comment:
Is there a specific reason why we do not use the `dates_active` decorator for those functions, yet?

Reply:
I generally like to split a new feature and associated refactorings into two PRs for several reasons:

1. The new feature can immediately be used in new code.
2. A long-running branch (new feature) that touches many files (refactorings) inevitably leads to merge conflicts.
3. A new feature requires an in-depth review, while a review of a manual refactoring is more about finding copy & paste errors.

The three functions I changed are just to verify that the decorator works as expected in the existing code.

Reply:
Makes a lot of sense. Thanks for the explanation!